test/integration/scheduler_perf/scheduler_bench_test.go

```diff
@@ -17,44 +17,87 @@ limitations under the License.
 package benchmark
 
 import (
+	"fmt"
 	"testing"
 	"time"
 
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/kubernetes/pkg/kubelet/apis"
 	"k8s.io/kubernetes/test/integration/framework"
 	testutils "k8s.io/kubernetes/test/utils"
 
 	"github.com/golang/glog"
 )
 
-// BenchmarkScheduling100Nodes0Pods benchmarks the scheduling rate
-// when the cluster has 100 nodes and 0 scheduled pods
-func BenchmarkScheduling100Nodes0Pods(b *testing.B) {
-	benchmarkScheduling(100, 0, b)
+// BenchmarkScheduling benchmarks the scheduling rate when the cluster has
+// various quantities of nodes and scheduled pods.
+func BenchmarkScheduling(b *testing.B) {
+	tests := []struct{ nodes, pods int }{
+		{nodes: 100, pods: 0},
+		{nodes: 100, pods: 1000},
+		{nodes: 1000, pods: 0},
+		{nodes: 1000, pods: 1000},
+	}
+	setupStrategy := testutils.NewSimpleWithControllerCreatePodStrategy("rc1")
+	testStrategy := testutils.NewSimpleWithControllerCreatePodStrategy("rc2")
+	for _, test := range tests {
+		name := fmt.Sprintf("%vNodes/%vPods", test.nodes, test.pods)
+		b.Run(name, func(b *testing.B) { benchmarkScheduling(test.nodes, test.pods, setupStrategy, testStrategy, b) })
+	}
 }
 
-// BenchmarkScheduling100Nodes1000Pods benchmarks the scheduling rate
-// when the cluster has 100 nodes and 1000 scheduled pods
-func BenchmarkScheduling100Nodes1000Pods(b *testing.B) {
-	benchmarkScheduling(100, 1000, b)
-}
+// BenchmarkSchedulingAntiAffinity benchmarks the scheduling rate when the
+// cluster has various quantities of nodes and scheduled pods.
+// New pods are scheduled with anti-affinity matching the pre-existing pods.
+func BenchmarkSchedulingAntiAffinity(b *testing.B) {
+	tests := []struct{ nodes, pods int }{
+		{nodes: 500, pods: 50},
+		{nodes: 500, pods: 400},
+	}
+	setupStrategy := testutils.NewSimpleWithControllerCreatePodStrategy("rc")
+	// NewSimpleWithControllerCreatePodStrategy will add name=rc to pods.
```

Comment:
Please put the comment above the function call.

Reply:
done